src/runtime/recipe/recipe-resolver.ts

```diff
@@ -159,7 +159,7 @@ export class ResolveWalker extends RecipeWalker {
 
 export class ResolveRecipeAction extends Action<Recipe> {
 
-  async generate(inputParams) {
+  async generate(inputParams: GenerateParams<Recipe>) {
```

Comment:
: StrategyParams

Reply:
I'm unsure about this, as ResolveRecipeAction is not a subclass of Strategy but a sibling (both are subclasses of Action<Recipe>). If we were to add more requirements to the StrategyParams we'd have to update the ResolveRecipeAction class even though it isn't necessarily effected.